draft-reddy-cose-jose-pqc-hybrid-hpke.md

```diff
@@ -113,6 +113,40 @@ For readability the algorithm ciphersuites labels are built according to the fol
 
 The HPKE PQ/T hybrid ciphersuites for JOSE and COSE are defined in {{IANA}}. Note that the PQ/T Hybrid KEM in HPKE is not an authenticated KEM. The HPKE Base mode can only be supported with the PQ/T Hybrid KEM.
 
+
+# AKP Key for X-Wing 
+
+This section describes the required parameters for an "AKP" key type, as defined in {{!I-D.ietf-cose-dilithium}}, and its use with the X-Wing algorithms, as defined in {{IANA}}. An example JWK is also provided for illustration.
+
+### Required Parameters
+
+A JSON Web Key (JWK) or COSE_Key with a key type ("kty") for use with the "X-Wing" algorithm includes the following parameters:
+
+- kty (Key Type)  
+  The key type parameter MUST be present and set to "AKP".  
+
+- alg (Algorithm)  
+  The algorithm parameter MUST be present and it MUST represent the X-Wing algorithm, as defined in {{IANA}}.  
+
+- pub (Public Key)  
+  The public key parameter MUST be present and contain the public key material in a format defined by the "AKP" key type.  
```

Comment:
AKP is a new key type that is like OKP, but does not require curve.

The spec that defines the "alg" can be as specific about "pub" and "priv" as you want it to be.
I recommend citing https://datatracker.ietf.org/doc/html/draft-connolly-cfrg-xwing-kem-06#name-key-generation

And stating that "pub" is concat(pk_M, pk_X)

In the case of JWK, this value is base64url encoded.

Reply:
Section 5.8.2. of draft-connolly-cfrg-xwing-kem-06 says:

"The publicKey component MUST be absent." (in private keys). That requirement is there for security reasons (as some stated security properties fail otherwise).

And "pub" is X-Wing public key. Treating the value as any sort of concatenation is asking for trouble.

Reply:
I modified the text to say "pub" is a X-Wing encapsulation key (pk), no need to explicitly say public key must be absent in private keys.

Reply:
OKP private keys have both x and d... AKP keys have both pub and priv.
At least that is the intention.

Reply:
@ilaril I tried to apply your suggestions here: https://github.com/tireddy2/Hybrid-KEM-with-COSE-JOSE/pull/5/files#r1876742416